Lib/shelve.py

```diff
@@ -56,13 +56,18 @@
 the persistent dictionary on disk, if feasible).
 """
 
-from pickle import DEFAULT_PROTOCOL, Pickler, Unpickler
+from pickle import DEFAULT_PROTOCOL, Unpickler, dumps, loads
```

Comment:
I don't think you need `Unpickler` now. Without it, you can also delay the import to `pickle` when you specify custom serializer / deserializer and explicit protocol.